File: src/models/Ability.ts
import {Activation} from "./Talent";
import Modifier from "./common/Modifier";
import Cost from "./common/Cost";
import Limit from "./common/Limit";

export default interface Ability {
    name: string
    description: string
    activation: Activation
    cost: Cost
    limiter: Limit
    modifiers: Modifier[]
}

File: gradle-java-server/src/main/java/com/github/genraven/gradlejavaserver/domain/Ability.java
package com.github.genraven.gradlejavaserver.domain;

import com.github.genraven.gradlejavaserver.domain.modifier.Modifier;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class Ability {
    private String name;
    private String description;
    private Activation activation;
    private Cost cost;
    private Limit limit;
    private List<Modifier> modifiers = new ArrayList<>();
}
